### ISSUE TITLE
rest api with flask

### ISSUE THREAD

**Issue by ghost** [abhijitpai000/customer_lifetime_value#1]
Hi @abhijitpai000 ,

Hope you are all well !

I was wondering if we could not create a flask rest api for your repository, it woud be awesome to compute data from e-commerce cms like Prestashop or Magento2.

What do you think about it ? How complicated it would be ?

Thanks for any insights or inputs on that.

Cheers,
Luc Michalski

**Comment by abhijitpai000** [comment]
Hi @lucmichalski, I'm not quite sure if I understand your request entirely but I'll try to answer. 

- You can create a rest API using the pickle file customer_lifetime_estimator in the src/models/ directory of the repo, feel free to utilize the implementation of using the pickle file in the src/predict.py to make the estimates.
- The implementation in this repo is specific to the Olist Public dataset, you'd have to work around that for Prestashop or Magneto2 data.

I hope this answers your questions, and feel free to let me know if you'd like to collaborate on creating the rest api or implementing CLV for Prestashop or Magento2.

Regards,
Abhijit Pai

**Comment by ghost** [comment]
@abhijitpai000 yeah, that would be awesome if could dockerize a rest api and create a prestashop module.

I can create the module :-)
